configure

```diff
@@ -672,41 +680,25 @@ def configure_node(o):
     o['variables']['node_target_type'] = 'static_library'
 
 
-def configure_libz(o):
-  o['variables']['node_shared_zlib'] = b(options.shared_zlib)
-
-  if options.shared_zlib:
-    o['libraries'] += ['-l%s' % options.shared_zlib_libname]
-  if options.shared_zlib_libpath:
-    o['libraries'] += ['-L%s' % options.shared_zlib_libpath]
-  if options.shared_zlib_includes:
-    o['include_dirs'] += [options.shared_zlib_includes]
-
-
-def configure_http_parser(o):
-    o['variables']['node_shared_http_parser'] = b(options.shared_http_parser)
-
-    if options.shared_http_parser:
-      o['libraries'] += ['-l%s' % options.shared_http_parser_libname]
-    if options.shared_http_parser_libpath:
-        o['libraries'] += ['-L%s' % options.shared_http_parser_libpath]
-    if options.shared_http_parser_includes:
-        o['include_dirs'] += [options.shared_http_parser_includes]
-
-
-def configure_libuv(o):
-  o['variables']['node_shared_libuv'] = b(options.shared_libuv)
-
-  if options.shared_libuv:
-    o['libraries'] += ['-l%s' % options.shared_libuv_libname]
-  if options.shared_libuv_libpath:
-    o['libraries'] += ['-L%s' % options.shared_libuv_libpath]
-  else:
-    o['variables']['uv_library'] = 'static_library'
-
-  if options.shared_libuv_includes:
-    o['include_dirs'] += [options.shared_libuv_includes]
-
+def configure_library(lib, output):
+  shared_lib = 'shared_%s' % lib
+  output['variables']['node_%s' % shared_lib] = b(getattr(options, shared_lib))
```

Comment:
Just a note but you can write this as `'shared_' + lib` and `'node_' + shared_lib` as well.  I'll leave it up to you to decide what's more readable.